src/libclient/utils/identicon.cpp

```diff
@@ -29,12 +12,26 @@ QImage make_identicon(const QString &name, const QSize &size)
 	QImage image(size, QImage::Format_ARGB32_Premultiplied);
 	image.fill(color);
 
+	int len = 0;
```

Comment:
This does handle grapheme clusters (this is the scan for joining characters), but despite knowing of `\p{xx}` since forever I somehow did not realise `\X` was a thing, which is obviously better and I will change this to do that.

Reply:
Ah, right, the combining characters are trailing. I had in mind it was the other way round since that's how they're typed. But yeah, just letting the regex do it is simpler either way.

Reply:
I have replaced this with the regexp.